xmtp_mls/src/groups/sync.rs

```diff
@@ -1309,6 +1350,22 @@ fn get_removed_leaf_nodes(
         .collect()
 }
 
+fn get_and_clear_pending_commit(
+    openmls_group: &mut OpenMlsGroup,
+    provider: &XmtpOpenMlsProvider,
+) -> Result<Option<Vec<u8>>, GroupError> {
+    // TODO: remove clone
+    if let Some(commit) = openmls_group.clone().pending_commit() {
```

Comment:
Wondering if we can use the commit as a reference rather than cloning the group?

Reply:
Compiler didn't like that because `openmls_group` is a mutable reference. I'm sure there's some hack to make it work.